modules/core/src/main/scala/org/tesselation/domain/cluster/programs/Joining.scala

```diff
@@ -72,13 +63,15 @@ object Joining {
       p2pClient,
       cluster,
       session,
+      sessionStorage,
+      selfId,
       joiningQueue
     ) {}
 
     def join: Pipe[F, P2PContext, Unit] =
       in =>
         in.evalMap { peer =>
-          joining.twoWayHandshake(peer) >>
+          joining.twoWayHandshake(peer, none) >>
```

Comment:
So we trust that the node we were connecting to initially, had the remoteAddress for other nodes correct right?